tests/steps/field/test_field_pack.py

```diff
@@ -0,0 +1,75 @@
+from frictionless import Resource, transform, steps
+
+# Issues
+
+
+def test_step_field_pack_header_907():
+    source = Resource("data/transform.csv")
+    target = transform(
+        source,
+        steps=[steps.field_pack(name="details", from_names=["name", "population"])],
+    )
+    assert target.schema == {
+        "fields": [{"name": "id", "type": "integer"}, {"name": "details"}]
+    }
+
+
+def test_step_field_pack_preserve_header_907():
+    source = Resource("data/transform.csv")
+    target = transform(
+        source,
+        steps=[
+            steps.field_pack(
+                name="details", from_names=["name", "population"], preserve=True
+            )
+        ],
+    )
+    assert target.schema == {
+        "fields": [
+            {"name": "id", "type": "integer"},
+            {"name": "name", "type": "string"},
+            {"name": "population", "type": "integer"},
+            {"name": "details"},
+        ]
+    }
+
+
+def test_step_field_pack_907():
+    source = Resource("data/transform.csv")
+    target = transform(
+        source,
+        steps=[
+            steps.field_pack(
+                name="details",
+                from_names=["name", "population"],
+                preserve=True,
+            )
+        ],
+    )
+    assert target.read_rows()[0] == {
+        "id": 1,
+        "name": "germany",
+        "population": 83,
+        "details": ["germany", "83"],
+    }
+
+
+def test_step_field_pack_object_907():
+    source = Resource("data/transform.csv")
+    target = transform(
+        source,
+        steps=[
+            steps.field_pack(
+                name="details",
+                from_names=["name", "population"],
+                field_type="object",
+                preserve=True,
+            )
+        ],
+    )
+    assert target.read_rows()[0] == {
+        "id": 1,
+        "name": "germany",
+        "population": 83,
+        "details": {"name": "germany", "population": "83"},
```

Comment:
We need to pack it as an array by default:
`"details": ["germany", "83"]`

Also, we can implement an object mode of packing also

Reply:
I have added both functions array and object mode. If we set data_type to object it packages as json and by default it does package as list ["germany", "83"]

Reply:
Now I see! Silly me :smile: